torchmetrics/functional/text/mer.py

```diff
@@ -63,26 +63,26 @@ def _mer_compute(errors: Tensor, total: Tensor) -> Tensor:
 
 
 def match_error_rate(
-    predictions: Union[str, List[str]],
-    references: Union[str, List[str]],
+    preds: Union[str, List[str]],
+    target: Union[str, List[str]],
```

Comment:
as it is breaking change we need to include depreciation in all public methods nad functions (not protected starting with _)
This means that we need to preserve the original argument at the end and set it as default None.
```suggestion
    preds: Union[str, List[str]],
    target: Union[str, List[str]],
    predictions: Union[None, str, List[str]] = None,
    references: Union[None, str, List[str]] = None,
```
and later in the beginning of the function add for each argument something like this:
```py
if predictions is None:
    warn("You are using deprecated argument `predictions` in v0.7 which was renamed to `preds`. "
             " The past argument will be removed in v0.8.",
             DeprecationWarning)
```

Reply:
to be completely compatible we would probably need to set all arguments as None by default as this will still fail if a user uses passing `match_error_rate(predictions="blnlfslncvlsvncl", references="fvdsbsdgbsdb")` and later in function add assert for each pair:
```py
assert preds is not None or predictions is not None
```

Reply:
personally, I would keep it easy and would not do the None for all by default...

Reply:
@ashutoshml otherwise great Work! lets just finish it :rabbit:

Reply:
Ok. Will include this in all public methods and functions and update the PR.

Reply:
> personally, I would keep it easy and would not do the None for all by default...

@Borda: What do you mean by this? Should we include edge cases for None or just keep them as is and add deprecation warning and relevant arguments?

Reply:
I mean that we need to make https://github.com/PyTorchLightning/metrics/pull/727#discussion_r780101069
but I don't think that this is needed https://github.com/PyTorchLightning/metrics/pull/727#discussion_r780102890

Reply:
@Borda I see. But if we don't do `None` for all, that would result in an error as pointed out by you [here](https://github.com/PyTorchLightning/metrics/pull/727#discussion_r780102890 ). I'll change and update the code, for now, let me know if there is another way.

Reply:
I know but I would bet on users that they use rather positional arguments than key-ward arguments...
for sure we can do also the None way, it is just ore work :rabbit: 
cc: @stancld @SkafteNicki @justusschock @PyTorchLightning/core-metrics

Reply:
Ok. In any case, I have updated the code to add warnings. I'll update the files based on the suggestions.

Reply:
I agree with Borda that users rather lean to use positional arguments. If keywords arguments were used then I'd assume one will use `preds, target` as indicated in a doc example. So I'd go with no `None` for `preds, target`, but I'm open to both solutions :]

Reply:
But then would it make sense to add `warnings` since they won't be accessible anyway?

Reply:
@ashutoshml .. Mmm, I've been thinking about that, and maybe `None` also for `preds, target` make more sense. Otherwise, as it's been already pointed out, when a user goes with keyword args, then empty `preds, target` will result in the error.

Reply:
yep I know the question is how much do we want to be to the user and how well back-compatible we want to be...
As mentioned I am fine with both ways, it would be just better I both your PRs will be coherent so if one goes with None, the other shall do the same... :rabbit:

Reply:
@stancld Let's go with `None` for `preds, target` too. I'll make the changes and update the PR. Does that sound good ?

Reply:
seems @stancld did it for his :rabbit:

Reply:
@ashutoshml Please, set even `preds, target` to None and add `type: ignore` where needed to be okay with `mypy`.